lib/src/html/html_generator_instance.dart

```diff
@@ -19,29 +19,17 @@ import 'resources.g.dart' as resources;
 import 'template_data.dart';
 import 'templates.dart';
 
-class HtmlGeneratorInstance implements HtmlOptions {
+typedef void FileWriter(String path, Object content, {bool allowOverwrite});
+
+class HtmlGeneratorInstance {
   final HtmlGeneratorOptions _options;
   final Templates _templates;
   final Package _package;
-  final String _outputDirectoryPath;
   final List<ModelElement> _documentedElements = <ModelElement>[];
-  final StreamController<File> _onFileCreated;
-
-  @override
-  String get relCanonicalPrefix => _options.relCanonicalPrefix;
```

Comment:
nice, I had vaguely remember wondering why we were doing this once, good to see it gone